### PR TITLE
FIX: Final fix for joint_mutect2 tumor_only in mutect2 subworkflow

### PR DESCRIPTION
**Pull request by maxulysse**

### CODE REVIEW

File: subworkflows/local/bam_variant_calling_tumor_only_all/main.nf [hunk 1]
```diff
@@ -113,12 +113,7 @@ workflow BAM_VARIANT_CALLING_TUMOR_ONLY_ALL {
     if (tools.split(',').contains('mutect2')) {
         BAM_VARIANT_CALLING_TUMOR_ONLY_MUTECT2(
             // Adjust meta.map to simplify joining channels
-            // joint_mutect2 mode needs different meta.map than regular mode
-            cram.map{ meta, cram, crai ->
-                joint_mutect2 ?
-                [ meta + [ id:meta.patient ] - meta.subMap('sample', 'status', 'data_type', 'patient') , cram, crai ] :
-                [ meta + [ id:meta.sample ] - meta.subMap('sample', 'status', 'data_type'), cram, crai ]
-            },
+            cram.map{ meta, cram, crai -> [ meta + [ id:meta.sample ] - meta.subMap('sample', 'status', 'data_type'), cram, crai ] },
```
Comment (FriederikeHanssen) [inline subworkflows/local/bam_variant_calling_tumor_only_all/main.nf:116]: so here you assign the sample name as Id for either scenario and in the subworkflow you then switch to patient if joint. why did it not pick up the conditional before?

Comment (maxulysse) [reply subworkflows/local/bam_variant_calling_tumor_only_all/main.nf:116]: Because we actually ran mpileup in regular mode

Comment (FriederikeHanssen) [reply subworkflows/local/bam_variant_calling_tumor_only_all/main.nf:116]: you pileupsummaries?

Comment (maxulysse) [reply subworkflows/local/bam_variant_calling_tumor_only_all/main.nf:116]: GETPILEUPSUMMARIES is run on each individual sample while MUTECT2 is ran jointly on all samples from the same patient.

Comment (FriederikeHanssen) [reply subworkflows/local/bam_variant_calling_tumor_only_all/main.nf:116]: 👍 good then we ar eon the same page :D

**Review by FriederikeHanssen — APPROVED** [review]

**Pull request merged**